packages/collector/src/index.js

```diff
@@ -94,32 +92,33 @@ function init(_config) {
   }
   // @ts-ignore: Property '__INSTANA_INITIALIZED' does not exist on type global
   global.__INSTANA_INITIALIZED = true;
-  config = normalizeConfig(_config);
+
+  /** @type {import('@instana/core/src/core').GenericLogger} */
+  logger = log.init(_config);
+  config = normalizeCollectorConfig(_config);
+  config = instanaNodeJsCore.util.normalizeConfig(config, logger);
 
   agentConnection = require('./agentConnection');
   const agentOpts = require('./agent/opts');
   const pidStore = require('./pidStore');
   const uncaught = require('./uncaught');
+  const announceCycle = require('./announceCycle');
+  const metrics = require('./metrics');
 
-  let logger;
-  logger = log.getLogger('index', newLogger => {
-    logger = newLogger;
-  });
-
-  // NOTE: By default we set our instana internal logger
-  config.logger = logger;
-
+  pidStore.init(config);
```

Comment:
NOTE: We define & init the pidstore in the parent module (collector) and pass it on to all modules, who use the pidstore. Dependency injection.